### PR TITLE
Fix : Mobile view navigation improvements

### PR DESCRIPTION
**Pull request by Abhinavxox**
### Description:
This pull request solves the following issues:
- In mobile view, the languages and tags picker section is collapsible
- In mobile view, when a language or tag is clicked it automatically scrolls down to the repositories list section

### Screenshots:

[Mobile View Improvement.webm](https://github.com/lucavallin/first-issue/assets/72064600/7c156e3b-be97-4715-8864-e5aaa89b2739)

### CODE REVIEW

File: components/NewsletterSection.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ export const NewsletterSection = () => {
   });
 
   return (
-    <div className="pt-6">
+    <div className="pt-6 hidden md:block">
```
Comment (lucavallin) [inline components/NewsletterSection.tsx:21]: let's show it on mobile too

File: components/Picker/LanguagePicker.tsx [hunk 2]
```diff
@@ -9,17 +12,38 @@ type LanguagePickerProps = {
...
+
+  const toggleCollapsible = () => {
+    setIsCollapsed(!isCollapsed);
+  };
+
   return (
     <div className="pt-6">
-      <SectionTitle text="Browse by Language" />
-      <div>
+      <div
```
Comment (lucavallin) [inline components/Picker/LanguagePicker.tsx:23]: let's hide it on `md`

File: components/Picker/SortPicker.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ export const SortPicker = ({ activeSort, sortOptions, onSortOrderSelect }: SortP
               className={`group mx-1 my-1 inline-block rounded-sm border px-2 py-1 text-sm ${
                 activeSort === sortOption
                   ? "active-pill"
-                  : "border-secondary transition-all transition-all hover:border-primary hover:text-primary"
```
Comment (lucavallin) [inline components/Picker/SortPicker.tsx:25]: Good catch!

File: components/Picker/TagPicker.tsx [hunk 4]
```diff
@@ -30,15 +37,30 @@ export const TagPicker = ({ tags, activeTagId, onTagPage }: TagPickerProps) => {
 
   return (
     <div className="pt-6">
-      <SectionTitle text="Browse by tag" />
-      <div>
+      <div
+        onClick={toggleCollapsible}
+        className={`flex cursor-pointer ${isCollapsed ? "sm:flex" : ""}`}
```
Comment (lucavallin) [inline components/Picker/TagPicker.tsx:42]: Let's hide it on `md`

File: components/Picker/TagPicker.tsx [hunk 5]
```diff
@@ -30,15 +37,30 @@ export const TagPicker = ({ tags, activeTagId, onTagPage }: TagPickerProps) => {
...
+        } ${isCollapsed ? "sm:max-h-full" : ""}`}
+      >
         {tags.slice(0, limit).map((tag) => {
           return (
             <PickerItem
               className={`group mx-1 my-1 inline-block rounded-sm border px-2 py-1 text-sm ${
                 onTagPage && tag.id === activeTagId
                   ? "active-pill"
-                  : "border-secondary transition-all transition-all hover:border-primary hover:text-primary"
+                  : "border-secondary transition-all hover:border-primary hover:text-primary"
```
Comment (lucavallin) [inline components/Picker/TagPicker.tsx:63]: Good catch!

**Comment by lucavallin** [comment]
Great contribution, thanks! I have left a few comments. If you address those, this can be merged!

**Review by lucavallin — APPROVED** [review]
Looks great - thank you very much!

**Comment by Abhinavxox** [comment]
> Looks great - thank you very much!

Thank you. Glad to contribute

**Pull request merged**